### PR TITLE
Adjust return type of `registry.find`

### PR DESCRIPTION
**Pull request by thomashacker**
## Description

Adjusts the return type of `registry.find` from `Callable` to `Dict[str, Optional[Union[str, int]]]` and code documentation so that it matches with `catalogue.registry.find`. 

### Types of change

bug fix

## Checklist

- [x] I confirm that I have the right to submit this contribution under the project's MIT license.
- [x] I ran the tests, and all new and existing tests passed.
- [x] My changes don't require a change to the documentation, or if they do, I've added all required information.

### CODE REVIEW

File: spacy/cli/package.py [hunk 1]
```diff
@@ -252,7 +252,7 @@ def get_third_party_dependencies(
                     raise regerr from None
             module_name = func_info.get("module")  # type: ignore[attr-defined]
             if module_name:  # the code is part of a module, not a --code file
-                modules.add(func_info["module"].split(".")[0])  # type: ignore[index]
+                modules.add(func_info["module"].split(".")[0])  # type: ignore
```
Comment (rmitsch) [inline spacy/cli/package.py:255]: Is there a specific `ignore` directive we can use here? Why does `[index]` not apply anymore?

File: spacy/util.py [hunk 2]
```diff
@@ -144,8 +144,17 @@ def get(cls, registry_name: str, func_name: str) -> Callable:
         return func
 
     @classmethod
-    def find(cls, registry_name: str, func_name: str) -> Callable:
-        """Get info about a registered function from the registry."""
+    def find(
+        cls, registry_name: str, func_name: str
+    ) -> Dict[str, Optional[Union[str, int]]]:
+        """Find the information about a registered function, including the
```
Comment (rmitsch) [inline spacy/util.py:150]:
```suggestion
        """Find information about a registered function, including the
```

**Review by rmitsch — COMMENTED** [review]
Minor questions/suggestions, but LGTM.

File: spacy/cli/package.py (hunk 1 above)
Comment (thomashacker) [reply spacy/cli/package.py:255]: It's throwing an `[union-index]` error because of the newly added `Union[str,int,None]`, I've adjusted the directive. 😄

Comment (rmitsch) [reply spacy/cli/package.py]: Thanks for checking!

Comment (rmitsch) [reply spacy/cli/package.py]: @thomashacker Wait, the code says `[union-attr]`, you mention `[union-index]` in your comment. Just making sure that `[union-attr]` is the correct one - is it?

Comment (thomashacker) [reply spacy/cli/package.py]: Oh sorry! I mixed them up somehow. I meant `[union-attr]` 😄

**Review by rmitsch — APPROVED** [review]

**Review by svlandeg — APPROVED** [review]
Good catch, thanks!

**Pull request merged**